File: client/src/app/modules/miscellaneous/services/billing.type.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { BillingType } from '../../miscellaneous/model/BillingType';

@Injectable({
  providedIn: 'root'
})
export class BillingTypeService extends AppRestService<BillingType>{

  constructor(http: HttpClient) {
    super(http, "miscellaneous");
  }

  getBillingTypes() {
    return this.getListCached(new HttpParams(), "billing-types");
  }

  addOrUpdateBillingType(billingType: BillingType) {
    this.clearListCache(new HttpParams(), "billing-types");
    return this.addOrUpdate(new HttpParams(), "billing-type", billingType, "Enregistré", "Erreur lors de l'enregistrement");
  }

}

File: src/main/java/com/jss/jssbackend/modules/miscellaneous/service/BillingTypeService.java
package com.jss.jssbackend.modules.miscellaneous.service;

import java.util.List;

import com.jss.jssbackend.modules.miscellaneous.model.BillingType;

public interface BillingTypeService {
    public List<BillingType> getBillingTypes();

    public BillingType getBillingType(Integer id);
}
